src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -404,6 +418,47 @@ def setPlineMarker(self, wflow, pName):
                 wflow['CleanupStatus'][pName] = False
         return wflow
 
+    def _getStatusTransitionTime(self, wflow, status=None):
+        """
+        A method to return status transition time for a workflow.
+        :param wflow:  A MSRuleCleanerWorkflow instance
+        :param status: The status transition to search for. If status is None
+                       then the last status transition time is returned
+        :return:       The UTC timestamp for the transition if it exists, None otherwise
+        """
+        try:
+            if status == None:
+                return wflow['RequestTransition'][-1]['UpdateTime']
+            else:
+                for trans in wfl['RequestTransition'][::-1]:
+                    if trans['Status'] == status:
+                        return trans['UpdateTime']
+                return None
+        except KeyError:
+            self.logger.debug("Could not find status transition history for %s", wflow['RequestName'])
+
+    def _isStatusAdvanceExpired(self, wflow, status=None):
```

Comment:
I would suggest to remove the `status` parameter here. You already pass the workflow information, so you can easily retrieve it through RequestStatus or RequestTransition.